q2_feature_classifier/_consensus_assignment.py

```diff
@@ -0,0 +1,229 @@
+            # the following allows dict lookup as str or int.
+            try:
+                t = ref_taxa[i[1]].split(t_delim)
+            except KeyError:
+                try:
+                    t = ref_taxa[int(i[1])].split(t_delim)
+                # if vsearch fails to find assignment, it reports '*' as the
+                # accession ID, which is completely useless and unproductive.
+                except ValueError:
+                    t = unassignable_label
+            if i[0] in obs_taxa.keys():
+                obs_taxa[i[0]].append(t)
+            else:
+                obs_taxa[i[0]] = [t]
+    return obs_taxa
+
+
+def _import_taxonomy_to_dict(infile):
+    ''' taxonomy file -> dict'''
+    with open(infile, "r") as inputfile:
+        lines = {line.strip().split('\t')[0]: line.strip().split('\t')[1]
+                 for line in inputfile}
+    return lines
+
+
+def _compute_consensus_annotations(query_annotations,
+                                   min_consensus,
+                                   t_delim=';',
+                                   unassignable_label="Unassigned"):
+    """
+        Parameters
+        ----------
+        query_annotations : dict of lists
+            Keys are query identifiers, and values are lists of all
+            taxonomic annotations associated with that identfier.
+        Returns
+        -------
+        dict
+            Keys are query identifiers, and values are the consensus of the
+            input taxonomic annotations.
+    """
+    # This code has been ported and adapted from QIIME 1.9.1 with
+    # permission from @gregcaporaso.
+    result = {}
+    for query_id, annotations in query_annotations.items():
+        consensus_annotation, consensus_fraction = \
+            _compute_consensus_annotation(annotations, min_consensus,
+                                          unassignable_label)
+        result[query_id] = (
+            t_delim.join(consensus_annotation), consensus_fraction)
+    return result
+
+
+def _compute_consensus_annotation(annotations, min_consensus,
+                                  unassignable_label):
+    """ Compute the consensus of a collection of annotations
+        Parameters
+        ----------
+        annotations : list of lists
+            Taxonomic annotations to compute the consensus of.
+        min_consensus : float
+            The minimum fraction of the annotations that a specfic annotation
+            must be present in for that annotation to be accepted. This must
+            be greater than or equal to 0.51.
+        unassignable_label : str
+            The label to apply if no acceptable annotations are identified.
+        Result
+        ------
+        consensus_annotation
+            List containing the consensus assignment
+        consensus_fraction
+            Fraction of input annotations that agreed at the deepest
+            level of assignment
+    """
+    # This code has been ported from QIIME 1.9.1 with
```

Comment:
Can you bump this up to just above the function definition (line 142 of this file)?